src/test/scala/chisel3/tests/ValidQueueTest.scala

```diff
@@ -65,24 +63,49 @@ class ValidQueueTest extends FlatSpec with ChiselScalatestTester {
     }
   }
 
-  it should "wait for data until valid data presented" in {
-    test(new ValidQueueModule(UInt(8.W), delay = 3)) { c =>
-      c.in.initSource().setSourceClock(c.clock)
-      c.out.initSink().setSinkClock(c.clock)
-
-      for(_ <- 0 until 10) {
-        c.out.expectInvalid()
-        c.clock.step()
+  it should "expectInvalid can be used to wait for data" in {
+    for(delay <- 1 until 4) {
+      test(new ValidQueueModule(UInt(8.W), delay = delay)) { c =>
+        c.in.initSource().setSourceClock(c.clock)
+        c.out.initSink().setSinkClock(c.clock)
+
+        for (_ <- 0 until 10) {
+          c.out.expectInvalid()
+          c.clock.step()
+        }
+
+        for (value <- 70 to 80) {
+          c.in.enqueueNow(value.U)
+          for (_ <- 0 until c.delay - 1) {
+            c.out.expectInvalid()
+            c.clock.step()
+          }
+          c.out.expectDequeueNow(value.U)
+        }
       }
+    }
+  }
 
-      c.in.enqueueNow(77.U)
-
-      c.out.expectInvalid()
-      c.clock.step()
-      c.out.expectInvalid()
-      c.clock.step()
-
-      c.out.expectDequeueNow(77.U)
+  it should "expectInvalid can be used to wait for data" in {
```

Comment:
wait how does this differ from the above test?

Reply:
Looks like it got duplicated. Removed copied code.